session/pingpong/consumer_balance_tracker.go

```diff
@@ -118,9 +133,46 @@ func (cbt *ConsumerBalanceTracker) Subscribe(bus eventbus.Subscriber) error {
 	if err != nil {
 		return err
 	}
+	err = bus.SubscribeAsync(pevent.AppTopicOrderUpdated, cbt.handleOrderUpdatedEvent)
+	if err != nil {
+		return err
+	}
 	return bus.SubscribeAsync(identity.AppTopicIdentityUnlock, cbt.handleUnlockEvent)
 }
 
+func (cbt *ConsumerBalanceTracker) handleOrderUpdatedEvent(ev pevent.AppEventOrderUpdated) {
+	if ev.Status.Incomplete() {
+		return
+	}
+
+	// pilvytis only allows topups on the default chain
```

Comment:
Topup requests are created using `FlagChainID` flag. So in theory you can create topup request on a different chains.
Pilvytis also allows a list chain id's to be used for topping up, (it's just that for now there is only one item in the list.)

So i guess at least we could use the flag here and not the default constant.